packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js

```diff
@@ -366,6 +366,56 @@ describe('Checkout', () => {
                 expect(spinner.exists()).toBe(true);
             });
         });
+
+        describe('isFormSubmitting', () => {
+            it('should display a loading spinner on the submit button when `isFormSubmitting` is true', () => {
+                // Arrange
+                const wrapper = mount(VueCheckout, {
+                    i18n,
+                    store: createStore(),
+                    localVue,
+                    propsData,
+                    data () {
+                        return {
+                            errorFormType: null,
+                            isFormSubmitting: true
+                        };
+                    }
+                });
+
+                // Act
+                const submitButton = wrapper.find('[data-test-id="confirm-payment-submit-button"]');
+                const spinner = submitButton.find('[data-test-id="action-button-spinner"]');
+
+                // Assert
+                expect(submitButton.attributes('aria-busy')).toBeTruthy();
+                expect(spinner.exists()).toBe(true);
+            });
+
+            it('should NOT display a loading spinner on the submit button when `isFormSubmitting` is false', () => {
+                // Arrange
+                const wrapper = mount(VueCheckout, {
+                    i18n,
+                    store: createStore(),
+                    localVue,
+                    propsData,
+                    data () {
+                        return {
+                            errorFormType: null,
+                            isFormSubmitting: false
+                        };
+                    }
+                });
+
+                // Act
+                const submitButton = wrapper.find('[data-test-id="confirm-payment-submit-button"]');
+                const spinner = submitButton.find('[data-test-id="action-button-spinner"]');
+
+                // Assert
+                expect(submitButton.attributes('aria-busy')).toBeFalsy();
+                expect(spinner.exists()).toBe(false);
+            });
+        });
```

Comment:
I personally don't think these unit tests should test what they are testing. `f-button`'s unit tests will cover that the spinner is displayed when it's loading. The same for the `aria-busy`.

I think we should be testing that `isFormSubmitting` is set to `true` and `false` within `onFormSubmit`, and that `isFormSubmitting` is used on the button correctly.

Then, we'd probably want to cover that the button in the `f-checkout` form is looking as expected with a Visual Regression Test.

Reply:
Yeah, it does make more sense and I wasn't really sure what level to test. So I will go with what you suggested.

On the visual regression test side of things, I was struggling to get the test to stay in a loading state because it seems the Percy configuration is controlled by props, like Storybook. In fact, I think it uses the knobs and URL parameters to control it. So unsure how I'll test a loading state.